components/remerge/src/storage/db.rs

```diff
@@ -143,16 +122,238 @@ impl RemergeDb {
                 :vclock,
                 :client_id
             )",
-            rusqlite::named_params! {
+            named_params! {
                 ":guid": id,
                 ":schema_ver": self.info.local.version.to_string(),
+                ":record": record,
                 ":now": now,
                 ":status": SyncStatus::New as u8,
                 ":vclock": vclock,
-                ":client_id": self.info.client_id,
+                ":client_id": self.client_id,
             },
         )?;
         tx.commit()?;
         Ok(id)
     }
+
+    fn counter_bump(&self) -> Result<Counter> {
+        use super::meta;
+        let mut ctr = meta::get::<i64>(&self.db, meta::CHANGE_COUNTER)?;
+        assert!(
+            ctr >= 0,
+            "Corrupt db? negative global change counter: {:?}",
+            ctr
+        );
+        ctr += 1;
+        meta::put(&self.db, meta::CHANGE_COUNTER, &ctr)?;
```

Comment:
It occurs to me... is this re-entrant?

Reply:
No, that two different things could call `counter_bump` at the same time and cause strange results (the same counter twice, a counter not being bumped as many times as there are calls, that sort of thing).